### PR TITLE
chore: update gatekeeper from 3.0.2-beta.1 to 3.2.2

### PR DESCRIPTION
**Pull request by GoelDeepak**
**What type of PR is this?**
Chore

**What this PR does/ why we need it**:
upgrade gatekeeper addon to newer version

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-73506

**Special notes for your reviewer**:
The upstream chart is located at https://github.com/open-policy-agent/gatekeeper/tree/master/charts/gatekeeper

I have done the manual upgrade test. Here are the steps that I followed:
```
1. Provisioned a Konvoy 1.6 cluster with gatekeeper enabled
2. Created a gatekeeper policy to validate allowed repository that a pod can use
     ( Here is the configuration that I used to test https://github.com/arbhoj/gatekeeper-policies/tree/master/allowed-repos-deployment-example)
3. Validated that the policy is working 
4. Upgraded gatekeeper using this chart
5. Validated that the policy is still working
```

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/gatekeeper/values.yaml [hunk 1]
```diff
@@ -19,19 +16,23 @@ commonLabels: {}
...
 
-image:
-  # image.name is the name of the Gatekeeper container image.
-  name: quay.io/open-policy-agent/gatekeeper
-  # image.tag is the tag name of the Gatekeeper container image.
-  tag:
-  # image.pullPolicy specifies when to pull the image.
-  pullPolicy: IfNotPresent
+## Number of replicas by default
+replicas: 1
```
Comment (hectorj2f) [inline staging/gatekeeper/values.yaml:20]: The default replicas are 3 in the chart repository 🤔 . I am wondering whether we need to deploy 2 at least.

**Review by hectorj2f — APPROVED** [review]
lgtm, i just added a comment but we could move forward to star testing in our builds.

File: staging/gatekeeper/Chart.yaml [hunk 2]
```diff
@@ -1,13 +1,12 @@
 apiVersion: v1
-appVersion: "3.0.4-beta.1"
+appVersion: "3.2.2"
 description: Gatekeeper - Policy Controller for Kubernetes
 name: gatekeeper
-version: 0.4.2
+version: 0.4.3
```
Comment (jimmidyson) [inline staging/gatekeeper/Chart.yaml:5]: Let's bump this
```suggestion
version: 0.5.0
```

File: staging/gatekeeper/README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ Kubernetes.
 
 ## Prerequisites
 
-- Kubernetes 1.9 (or newer) for validating and mutating webhook admission
+- Kubernetes 1.19 (or newer) for validating and mutating webhook admission
```
Comment (jimmidyson) [inline staging/gatekeeper/README.md:8]: Is this a typo? Assuming this does work on older versions of Kubernetes?

**Review by jimmidyson — APPROVED** [review]
A couple of nits and a question, but LGTM

**Review by makkes — APPROVED** [review]

File: staging/gatekeeper/README.md (hunk 3 above)
Comment (GoelDeepak) [reply staging/gatekeeper/README.md:8]: Good point! I checked and it seems they recommend 1.14 https://github.com/open-policy-agent/gatekeeper#minimum-kubernetes-version. I will update that

**Pull request merged**